src/ducks/SANCharts/Charts.js

```diff
@@ -177,6 +181,51 @@ class Charts extends React.Component {
     }
   }
 
+  setupDayMetrics = ({ chartData, isIntervalSmallerThanDay, metrics }) => {
+    const dayMetrics = []
+    let alignedChartData = chartData
+
+    if (isIntervalSmallerThanDay) {
+      metrics.forEach(
+        ({ key, minInterval }, index) =>
+          minInterval === '1d' && dayMetrics.push([key, index])
+      )
+
+      const dayStartMetrics = {}
+
+      alignedChartData = chartData.map(({ ...data }) => {
+        const { datetime: currentPointDatetime } = data
+
+        for (let i = 0; i < dayMetrics.length; i++) {
```

Comment:
foreach? 'i' is not using inside loop

Reply:
1. `LOC 200` -> `dayMetrics[i]`
2. Creating functions inside the loop is a bad idea. `chartData` usually has `150+` items, which will result in the same amount of **a new closures**